cristin-import/src/main/java/no/unit/nva/cristin/mapper/nva/ReferenceBuilder.java

```diff
@@ -138,7 +150,73 @@ private Anthology buildChapterForPublicationContext() {
     }
 
     private PublicationContext buildEventForPublicationContext() {
-        return new Event.Builder().build();
+        return new Event.Builder()
+                   .withLabel(extractLabel())
+                   .withAgent(extractAgent())
+                   .withPlace(extractPlace())
+                   .withTime(extractTime())
+                   .build();
+    }
+
+    private Time extractTime() {
+        var from = extractCristinEventFromDate();
+        var to = extractCristinEventToDate();
+        return nonNull(from) && nonNull(to) ? new Period(from, to) : null;
+    }
+
+    private Instant extractCristinEventFromDate() {
+        return Optional.ofNullable(cristinObject)
+                   .map(CristinObject::getLectureOrPosterMetaData)
+                   .map(CristinLectureOrPosterMetaData::getEvent)
+                   .map(no.unit.nva.cristin.mapper.Event::getFrom)
+                   .map(this::toInstant)
+                   .orElse(null);
+    }
+
+    private Instant extractCristinEventToDate() {
+        return Optional.ofNullable(cristinObject)
+                   .map(CristinObject::getLectureOrPosterMetaData)
+                   .map(CristinLectureOrPosterMetaData::getEvent)
+                   .map(no.unit.nva.cristin.mapper.Event::getTo)
+                   .map(this::toInstant)
+                   .orElse(null);
+    }
+
+    private Instant toInstant(String date) {
+        var formatter = DateTimeFormatter.ofPattern("yyyy-MM-dd'T'HH:mm:ss");
```

Comment:
This one is sketchy, but time format from cristin is not "standard" LocalDateTime